command/agent/rpc.go

```diff
@@ -616,6 +626,25 @@ func (i *AgentRPC) handleSerfQuery(client *rpcClient, seq uint64) error {
 	return client.Send(&resp, nil)
 }
 
+func (i *AgentRPC) handleSerfPing(client *rpcClient, seq uint64) error {
+	var req serfPingRequest
+	if err := client.dec.Decode(&req); err != nil {
+		return fmt.Errorf("decode failed: %v", err)
+	}
+
+	i.logger.Printf("[DEBUG] agent.rpc: handling serf ping request")
+
+	// Start the serf query
+	resp, err := i.agent.SerfPing(req.Name)
```

Comment:
Does it make sense to proceed from line 638 if there's an error setting up the serf ping query?  I was thinking we should just do an `if err return err` here.